examples/Main.hs

```diff
@@ -14,16 +14,23 @@ fact, fib :: Int -> Int
 fact n = if n == 0 then 1 else n * fact (n - 1)
 fib n = case n of 1 -> 1; 2 -> 1; _ -> fib (n - 1) + fib (n - 2)
 
+choose :: Int -> Int -> Int
+choose n k = fact n `div` (fact k * fact (n - k))
+
+
 benchmarks :: [Benchmark]
 benchmarks =
     [ bench "id" (nf id ())
     , series [10..10] $ \n ->
         bgroup "pure-functions"
-          [ bench "fact" (use n >>= nf fact)
-          , bench "fib" (use n >>= nf fib)
+          [ bench "fact" (nf fact n)
+          , bench "fib" (nf fib n)
           ]
     , withSampling (timebounded (repeat 10) fiveSecs) $ bgroup "roundrip"
         [ bench "ping" (nfIO (system "ping -c1 8.8.8.8 > /dev/null")) ]
+    , series [1..4] $ \n ->
+        series [1..n] $
+          bench "n choose k" . nf (n `choose`)
```

Comment:
This is in theory not the same as `nf (uncurry choose) (n, k)`, because `choose` could well be doing part of the computation already, which will then be memoized (e.g. if there's a `let` in the body of `choose` in between abstracting the first argument and abstracting the second).